lib/flutter_zoom_drawer.dart

```diff
@@ -160,89 +185,195 @@ class _ZoomDrawerState extends State<ZoomDrawer>
   final Curve _scaleDownCurve = const Interval(0.0, 0.3, curve: Curves.easeOut);
   final Curve _scaleUpCurve = const Interval(0.0, 1.0, curve: Curves.easeOut);
   final Curve _slideOutCurve = const Interval(0.0, 1.0, curve: Curves.easeOut);
-  final Curve _slideInCurve =
-      const Interval(0.0, 1.0, curve: Curves.easeOut); // Curves.bounceOut
+  final Curve _slideInCurve = const Interval(0.0, 1.0, curve: Curves.easeOut);
   ColorTween _overlayColor =
       ColorTween(begin: Colors.transparent, end: Colors.black38);
 
-  /// check the slide direction
+  static bool _shouldDrag = false;
+
+  /// Check the slide direction
   late int _rtlSlide;
 
-  AnimationController? _animationController;
-  DrawerState _state = DrawerState.closed;
+  late final AnimationController _animationController;
+  double get _animationValue => _animationController.value;
+
+  DrawerLastAction drawerLastAction = DrawerLastAction.closed;
+
+  /// Drawer state
+  final ValueNotifier<DrawerState> stateNotifier =
```

Comment:
this should be a private variable

Reply:
How the user will listen to stateNotifier?

Reply:
We will have a public getter that users can listen to